step/src/main/java/com/google/sps/Notification.java

```diff
@@ -35,6 +35,9 @@ public String getId() {
 
   // Method for getting the id of the user who sent the notification.
   public String getOtherUserId() {
+    if (otherUserId == null) {
```

Comment:
How could otherUserId be null here? I think a better approach is to reject a null otherUserId being passed in the constructor in the first place. Same thing with the id parameter.

Reply:
It technically shouldn't but I think my thought process while writing this was that you can never be overly cautious. I like the constructor suggestion though!